pkl-core/pkl-core.gradle.kts

```diff
@@ -273,11 +268,23 @@ val testAlpineExecutableAmd64 by tasks.registering(Test::class) {
 }
 
 tasks.testNative {
-  dependsOn(testLinuxExecutableAmd64)
-  dependsOn(testLinuxExecutableAarch64)
-  dependsOn(testMacExecutableAmd64)
-  dependsOn(testMacExecutableAarch64)
-  dependsOn(testAlpineExecutableAmd64)
+  when {
+    buildInfo.os.isMacOsX && buildInfo.arch == "aarch64" -> {
+      dependsOn(testMacExecutableAarch64)
+    }
+    buildInfo.os.isMacOsX && buildInfo.arch == "amd64" -> {
+      dependsOn(testMacExecutableAmd64)
+    }
```

Comment:
```suggestion
    buildInfo.os.isMacOsX -> {
      dependsOn(testMacExecutableAarch64)
      dependsOn(testMacExecutableAmd64)
    }
```

Reply:
Do you really want this by default, for every developer build? (You can still run the individual tasks if you need to.) Does it also work for Intel Macs?

Reply:
These tasks (`buildNative`, `testNative`) aren't part of the base Gradle lifecycle tasks (build, test); they don't get run unless explicitly run. I think it's fine to let `buildNative` do a cross-build on macOS; you can also selectively build an architecture by running one of the subtasks.

But yeah, good point on Intel; we should skip out on the aarch64 stuff there.

Reply:
I'd optimize for the most common dev need. But OK, made the requested change.

Reply:
Well, it's also consistent with how the linux/amd64 build works; on there, `buildNative` will build both libc and musl if it can.

Reply:
I'd definitely ask for a change if I could get the musl build to work. Until then I'm OK. :-)